### PR TITLE
feat(inventory groups): rename group modal

### PR DESCRIPTION
**Pull request by Fewwy**
Rename group modal for the groups table 
https://issues.redhat.com/browse/ESSNTL-3734

image
image
image
image

### CODE REVIEW

File: src/components/InventoryGroups/utils/api.js [hunk 1]
```diff
@@ -25,6 +25,13 @@ export const validateGroupName = (name) => {
     .then((resp) => resp?.results.some((group) => group.name === name));
 };
 
+export const updateGroupById = (id, payload) => {
+    return instance.put(`${INVENTORY_API_BASE}/groups/${id}`, {
+        Name: payload.name,
```
Comment (gkarat) [inline src/components/InventoryGroups/utils/api.js:30]: according to openapi there is only `name` field:
```suggestion
        name: payload.name,
```

File: src/components/InventoryGroups/Modals/RenameGroupModal.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            const results = await validateGroupName(value);
+            if (results === true) {
+                throw 'Group name already exists';
+            }
+
+            return undefined;
+        };
+
+        // eslint-disable-next-line new-cap
+        const d = awesomeDebouncePromise(check, 500, { onlyResolvesLast: false });
```
Comment (gkarat) [inline src/components/InventoryGroups/Modals/RenameGroupModal.js:62]: why we need `onlyResolvesLast` set to false?

File: src/components/InventoryGroups/Modals/RenameGroupModal.cy.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                }
+            }
+        });
+    });
+
+    beforeEach(() => {
+
+        cy.intercept('POST', '**/api/inventory/v1/groups', {
+            statusCode: 504
+        }).as('create_group');
```
Comment (gkarat) [inline src/components/InventoryGroups/Modals/RenameGroupModal.cy.js:69]: Why you need it here?

File: src/components/InventoryGroups/Modals/RenameGroupModal.cy.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        );
+    });
+
+    it('Input is fillable and firing a validation request that succeeds', () => {
+        cy.get(TEXT_INPUT).type('sre-group0');
+        cy.wait('@validate').then((xhr) => {
+            expect(xhr.request.url).to.contain('groups');}
+        );
+        cy.get(`button[type="submit"]`).should('have.attr', 'aria-disabled', 'true');
+    });
```
Comment (gkarat) [inline src/components/InventoryGroups/Modals/RenameGroupModal.cy.js:91]: Can we also check the successful scenario? When user puts a new name which doesn't yet exist and then click 'Rename'?

**Review by gkarat — CHANGES_REQUESTED** [review]
Other notes:

- Please reference Jira ID in your commit and PR's description.
- Cypress errors when running "after all" functions (see the attached video)

[Screencast from 02-20-2023 05:46:27 PM.webm](https://user-images.githubusercontent.com/31385370/220162750-8e6e0a5e-4c3e-4627-9084-a7cb74d93ee9.webm)

File: src/components/InventoryGroups/utils/api.js (hunk 1 above)
Comment (Fewwy) [reply src/components/InventoryGroups/utils/api.js:30]: Thanks, fixed

File: src/components/InventoryGroups/utils/api.js [hunk 5]
```diff
@@ -25,6 +25,13 @@ export const validateGroupName = (name) => {
...
+        Type: 'static'
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/components/InventoryGroups/utils/api.js:31]: Yep, mistake!

File: src/components/InventoryGroups/Modals/RenameGroupModal.cy.js (hunk 4 above)
Comment (Fewwy) [reply src/components/InventoryGroups/Modals/RenameGroupModal.cy.js:91]: Wrote a successful scenario with a intercept of PUT request

File: src/components/InventoryGroups/Modals/RenameGroupModal.js (hunk 2 above)
Comment (Fewwy) [reply src/components/InventoryGroups/Modals/RenameGroupModal.js:62]: According to https://www.npmjs.com/package/awesome-debounce-promise this option allows to resolve all promises that were returned and not the "only last returned". This was added in the create group modal - should we remove it?

**Comment by Fewwy** [comment]
@gkarat Updated the PR

File: src/components/InventoryGroups/Modals/RenameGroupModal.js (hunk 2 above)
Comment (gkarat) [reply src/components/InventoryGroups/Modals/RenameGroupModal.js:62]: @Fewwy why would you need to resolve all the promises? aren't we interested in only the last one?

**Review by gkarat — APPROVED** [review]

**Pull request merged**